js/backup.js

```diff
@@ -0,0 +1,659 @@
+                name,
+                ':',
+                error && error.stack ? error.stack : error
+              );
+              return reject(error);
+            });
+          }
+        };
+      });
+    });
+  }
+
+  function getConversationDirName(conversation) {
+    var name = conversation.active_at || 'never';
+    if (conversation.type === 'private') {
+      name += ' (' + (conversation.name || conversation.id) + ')';
+    } else {
+      name += ' (' + conversation.name + ')';
+    }
+    return name;
+  }
+
+  function getConversationLoggingName(conversation) {
+    var name = conversation.active_at || 'never';
+    name += ' (' + conversation.id + ')';
+    return name;
+  }
+
+  function exportConversations(idb_db, parentDir) {
+    return new Promise(function(resolve, reject) {
+      var transaction = idb_db.transaction('conversations', "readwrite");
+      transaction.onerror = function(e) {
+        console.log(
+          'exportConversations: transaction error:',
+          e && e.stack ? e.stack : e
+        );
+        return reject(e);
+      };
+      transaction.oncomplete = function() {
+        // not really very useful - fires at unexpected times
+      };
+
+      var promiseChain = Promise.resolve();
+      var store = transaction.objectStore('conversations');
+      var request = store.openCursor();
+      request.onerror = function(e) {
+        console.log(
+          'exportConversations: error pulling conversations:',
+          e && e.stack ? e.stack : e
+        );
+        return reject(e);
+      };
+      request.onsuccess = function(event) {
+        var cursor = event.target.result;
+        if (cursor && cursor.value) {
+          var conversation = cursor.value;
+          var dir = getConversationDirName(conversation);
+          var name = getConversationLoggingName(conversation);
+
+          var process = function() {
+            return createDirectory(parentDir, dir).then(function(dir) {
+              return exportConversation(idb_db, name, conversation, dir);
+            });
+          };
+
+          console.log('scheduling export for conversation', name);
+          promiseChain = promiseChain.then(process);
+          cursor.continue();
+        } else {
+          console.log('Done scheduling conversation exports');
+          return promiseChain.then(resolve, reject);
+        }
+      };
+    });
+  }
+
+  function getDirectory() {
+    return new Promise(function(resolve, reject) {
+      var w = extension.windows.getViews()[0];
+      if (w && w.chrome && w.chrome.fileSystem) {
+        w.chrome.fileSystem.chooseEntry({
+          type: 'openDirectory'
+        }, function(entry) {
+          if (!entry) {
+            var error = new Error('Error choosing directory');
+            error.name = 'ChooseError';
+            reject(error);
+          } else {
+            resolve(entry);
+          }
+        });
+      }
```

Comment:
No, there are only two options - `entry` is either truthy or falsey. I'll make a small edit to make it a little clearer.